hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractAddListenerMessageTask.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl.protocol.task;
+
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.internal.nio.Connection;
+
+import java.util.UUID;
+
+/**
+ * Base message task for listener registration tasks
+ */
+public abstract class AbstractAddListenerMessageTask<P>
```

Comment:
can be renamed as `AbstractDestroyableListenerMessageTask`?

Reply:
It adds listener so existing name sounds better.